File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/Notebook.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Notebook Declaration
//==============================================================
type Notebook struct {
    gorm.Model
     Title                                    string
    Repository                                                            string
    WorkspaceId         *uint
    Workspace           *AnalyticsWorkspace `gorm:"foreignKey:WorkspaceId"`
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromNotebookId"`
     Experiments           []Experiment `gorm:"foreignKey:ExperimentsFromNotebookId"`
     Queries           []BIQuery `gorm:"foreignKey:QueriesFromNotebookId"`
    Language                      NotebookLanguage

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/notebook.rb
class Notebook < ApplicationRecord
  enum Language: [:Python, :R, :SQL, :Julia]


  composed_of :repositoryRef,
    class_name: "RepositoryRef",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      %w[repositoryRef_path path]
    ]

  has_many :Workspace, class_name: 'AnalyticsWorkspace'
  has_many :Datasets, class_name: 'DataSet'
  has_many :Experiments, class_name: 'Experiment'
  has_many :Queries, class_name: 'BIQuery'

end
